python/packages/autogen-agentchat/src/autogen_agentchat/agents/_assistant_agent.py

```diff
@@ -233,6 +240,30 @@ def __init__(
         self._model_context: List[LLMMessage] = []
         self._is_running = False
 
+    def add_tool(self, tool: Tool | Callable[..., Any] | Callable[..., Awaitable[Any]]) -> None:
```

Comment:
@JMLX42 already has a PR on this. Let's move `add_tool` `remove_tool` out of this PR, and continue our discussion on #4545 ?

Reply:
Sounds good